pkg/agent/podresapi/resources.go

```diff
@@ -114,6 +114,16 @@ func (l *PodResourcesList) GetContainer(ns, pod, ctr string) *ContainerResources
 	return l.GetPodResources(ns, pod).GetContainer(ctr)
 }
 
+func (l *PodResourcesList) PurgePodResources(ns, pod string) {
+	if l == nil {
+		return
+	}
+
+	if podMap, ok := l.m[ns]; ok {
+		delete(podMap, pod)
+	}
```

Comment:
```suggestion
	if podMap, ok := l.m[ns]; ok {
		if _, exists := podMap[pod]; exists {
			delete(podMap, pod)
		}
	}
```
nit, perhaps it is safer to check of the Pod actually exists in a given ns before trying to delete from the map?

Reply:
Considering the golang docs/specs explicitly state for delete that ' If m is nil or there is no such element, delete is a no-op.', it is always better to do it this way. Otherwise you would always end up doing two map key lookups for a single successful delete and an failed key lookup attempt for a deletion attempt of a non-existing key, which is a clearly a net loss (compared to the current single lookup and a single lookup attempt).

Reply:
fair enough